core/src/main/java/com/datastax/oss/driver/api/core/cql/QueryTrace.java

```diff
@@ -43,6 +43,12 @@
   @NonNull
   InetAddress getCoordinator();
 
+  /**
+   * The port of the node that coordinated the query. Prior to C* 4.0 this is not set and will
+   * default to 0.
+   */
+  int getCoordinatorPort();
+
```

Comment:
> I took the approach of adding a separate port int.. another approach would have to used an InetSocketAddress and set the port there. However that would involve me modifying an existing interface method, instead of simply adding to one. Let me know what you think.

Agreed, but we can have the new method return `InetSocketAddress` if it has a different name. I suggest `getCoordinatorAddress` (for lack of a better idea). We can also make it default to avoid the breaking change:

```java
  @NonNull
  @Deprecated
  InetAddress getCoordinator();

  @NonNull
  default InetSocketAddress getCoordinatorAddress() {
    return new InetSocketAddress(getCoordinator(), 0);
  }
```
It would still be overridden in the default implementation, to return the correct port if available.

Reply:
I took your suggestion, and converted the port into an InetSocketAddress, I did this for both the TraceEvent and the QueryTrace. I was able to remove the revapi exception. Let me know if anything else needs to be addressed,  but I think this is buttoned up.